subprojects/core/src/main/java/org/gradle/internal/fingerprint/FingerprinterSpec.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.fingerprint;
+
+import org.gradle.api.tasks.FileNormalizer;
+
+/**
+ * Specifies criteria for selecting a {@link FileCollectionFingerprinter}
+ */
+public interface FingerprinterSpec {
```

Comment:
I think this would better be called `FileNormalizationSpec`. We currently only use "fingerprinting" for input _files,_ but this can change if we introduce normalization for input values.

I also dislike "fingerprinting" as a name, and would love to replace it with something else that is more different to "snapshotting." Thus it would be nice not to embed the word more in the codebase.

Reply:
Done